config/agent.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"flag"
+	"log"
+	"log/slog"
+	"os"
+	"time"
+
+	"github.com/caarlos0/env/v11"
+	"github.com/mihailtudos/metrickit/pkg/flags"
+)
+
+const DefaultReportInterval = 10
+const DefaultPoolInterval = 2
+
+var serverAddr *flags.ServerAddr
+var poolIntervalInSeconds *flags.DurationFlag
+var reportIntervalInSeconds *flags.DurationFlag
+
+type AgentConfig struct {
+	Log            *slog.Logger
+	ServerAddr     string
+	PollInterval   time.Duration
+	ReportInterval time.Duration
+}
+
+type EnvAgentConfig struct {
+	PollInterval   *int    `env:"POLL_INTERVAL"`
+	ReportInterval *int    `env:"REPORT_INTERVAL"`
+	ServerAddr     *string `env:"ADDRESS"`
+}
+
+func NewAgentConfig() *AgentConfig {
+	parseFlags()
+
+	return &AgentConfig{
+		PollInterval:   poolIntervalInSeconds.GetDuration(),
+		ReportInterval: reportIntervalInSeconds.GetDuration(),
+		ServerAddr:     serverAddr.String(),
+		Log:            slog.New(slog.NewJSONHandler(os.Stdout, nil)),
+	}
+}
+
+func parseFlags() {
+	var cfg EnvAgentConfig
+	if err := env.Parse(&cfg); err != nil {
+		log.Panic("failed to parse env vars: ", err.Error())
```

Comment:
See the video about error handling: https://app.pachca.com/chats/9180496?message=234290731